src/ai/defaultai.cc

```diff
@@ -6808,8 +6809,8 @@ void DefaultAI::review_wares_targets(const Time& gametime) {
 				default_target = kTargetQuantCap;
 			}
 
-			const uint16_t new_target = std::max<uint16_t>(default_target * multiplier / 10, 3);
-			assert(new_target > 1);
+			const uint64_t new_target = std::max<uint64_t>(multiplier * default_target / 10, 3);
+			assert(new_target > 1 && new_target < 30000);
```

Comment:
There seems to be no mechanism in place to ensure the target will be below 30000 what is the reason for this new part in the assert? why should it be necessary? In the old code this isn't  assured as well as there is a max of 2^16.

Reply:
I changed the assert to `new_target < std::numeric_limits<uint16_t>::max()`

Reply:
but we can't guarantee this assert as we widened the result above to uint64_t. Furthermore I don't get the sense of this assert. All I know is having an assert which can't be guaranteed feels not right. I believe they should be used to show whether something went wrong in our code that was not intended. And this is no case like this. 
best would be to just delete this part imho.